### PR TITLE
port GRPC to HPUX 11.31 OS (ia64)

### PR DESCRIPTION
**Pull request by malexzx**
Hi

this is my port GRPC to HPUX 11.31 OS.
This port assume to be compiled by gcc/g++ 4.8.5.

### CODE REVIEW

**Comment by malexzx** [comment]
changing email address to conform CLA

File: src/core/lib/support/murmur_hash.h [hunk 1]
```diff
@@ -38,6 +38,21 @@
 
 #include <stddef.h>
 
+#ifdef __hpux
```
Comment (murgatroid99) [inline src/core/lib/support/murmur_hash.h]: As in the other cases, this should use a constant defined in `port_platform.h`. In this particular case, I would recommend something like `GPR_LITTLE_ENDIAN`.

**Comment by malexzx** [comment]
Thanks for your review!
I have a question about what should I do to continue. 
>>This change will have significant conflicts with #8125. In particular, socklen_t will be removed from all >>internal headers (and many other places), and port_platform.h will be split, with part of it moved to 
>> src/core/lib/iomgr/port.h.
Wait until these changes (#8125) go to the main branch and then on this basis to create my changes?
Create changes based on #8125 branch now?

**Comment by malexzx** [comment]
What I sould do with a socklen_t within function body?  The socklen_t type appear in HPUX system headers but used implementation (BSD_SOCKETS) is required to use int. Define socklen_t to int won't work. Copy-paste whole functions, like grpc_set_socket_reuse_addr bump code to grow and not simplify.
Maybe good example is avoid use socklen_t at all? And use asserts and casts only before and for system call?
For example:
  ```
socklen_t intlen = sizeof(newval);
..................
  if (0 != getsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &newval, &intlen)) {
```
is changed to
 ```
 size_t intlen = sizeof(newval);
.............
 GPR_ASSERT(intlen < GRPC_SOCKLEN_T_MAX);
 if (0 != getsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &newval, (GRPC_SOCKLEN_T*)&intlen)) {

```

**Comment by nicolasnoble** [comment]
I would suggest grpc_socklen_t

**Comment by ctiller** [comment]
grpc_socklen_type: _t names are reserved for the compiler & library
implementations

**Comment by ctiller** [comment]
(or grpc_socklen)

**Comment by nicolasnoble** [comment]
Yes, sorry, without the _t, my bad.

**Comment by malexzx** [comment]
continue in #8572

**Pull request closed without merging**